### DISCUSSION TITLE
System.Data.Odbc and netstandard2.0

### DISCUSSION THREAD

**Discussion by MaceWindu** [dotnet/runtime#78550] [Q&A]
Just curious. `System.Data.Odbc` 7.0.0 dropped explicit `netcoreapp3.1` target but still ship `netstandard2.0` which makes you think you can use 7.0.0 with netcore 3.1. But trying to use `nestandard2.0` build just gives you PNSE in runtime.

Question is: what usage scenario reserved for `netstandard2.0` builds if it is not possible to use them in runtime?

**Comment by MaceWindu** [comment]
Same question for system.data.oledb

**Reply by danmoseley** [reply]
@ViktorHofer

**Comment by ViktorHofer** [comment]
Please see the discussion in https://github.com/open-telemetry/opentelemetry-dotnet/issues/3448 for a thorough answer.

Personally, I don't like the current behavior where a package communicates to support .NET Standard but then throws at runtime either. Admittedly, there are analyzers now that warn you at build time, but the overall issue around the .NET Standard support policy in our packages still stands.

cc @ericstj @andrewlock

**Reply by MaceWindu** [reply]
related issue: #898

**Reply by MaceWindu** [reply]
As I suspected, it looks like `nestandard2.0` needed as build dependency for other `netstandard2.0` libraries.

I see following potential solution to it. There should be an option (e.g. in nuspec) to mark `nestandardX` TFM as facade-only asset visible only to `netstandardX+` projects which will result in build error when used to build "real" TFMs like `netcoreapp3.1` (as direct or transitive dependency).
